pkg/yurthub/pki/pki.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pki
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/client-go/util/certificate"
+)
+
+// GenTGenTLSConfigUseCertMgrAndCertPool generates a TLS configuration
+// using the given certificate manager and x509 CertPool
+func GenTLSConfigUseCertMgrAndCertPool(
+	m certificate.Manager,
+	root *x509.CertPool) (*tls.Config, error) {
+	tlsConfig := &tls.Config{
+		// Can't use SSLv3 because of POODLE and BEAST
+		// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
+		// Can't use TLSv1.1 because of RC4 cipher usage
+		MinVersion: tls.VersionTLS12,
+		RootCAs:    root,
+		ClientAuth: tls.RequireAnyClientCert,
```

Comment:
I think that it's not need to verify the client certificate by yurthub server, so `ClientAuth` can be setup with `tls. NoClientCert `.  meanwhile, `tlsConfig.RootCAs` setting is not needed.